test/should_pass/bool.erl

```diff
@@ -5,3 +5,16 @@
 -spec b(boolean(), bool()) -> bool().
 b(B1, B2) ->
     B1 andalso B2.
+
+%% variable bindings should propagate from first to second arg of orelse
+-spec b2() -> boolean().
+b2() ->
+    begin
+        A = f(),
+        is_integer(A)
+    end
+        orelse
+        is_float(A).
+
+-spec f() -> number().
+f() -> 1.

```

Comment:
This test is great and reminds me we don't have that many tests checking that we get the scoping of variables correctly. Erlangs scoping is rather weird and inconsistent, and I'm sure there are places (just like this one) where I've gotten it wrong. This test is a good start. One could also imagine a test for `or` that makes sure that we *don't* propagate the variable bindings across the operator.

Reply:
Yes, Erlang scoping is sometimes surprising, and it's hard to come up with thorough testing. (Unfortunately I don't have time to write extensive tests for Gradualizer, instead when I hit an issue, I try to add at least one example testcase which fails without the fix, regression testing style) I think Gradualizer has scoping right in most of the cases, I didnt find any issue on real code-bases. This testcase is a bit contrived, I only found it based on your comment in the code. But I will quickly scan the code if I see any other suspicious places.

Reply:
Sorry, I didn't mean to imply that you had to write these test for this pull-request. I was merely thinking out loud about the lack of tests for the scoping rules. No need to write any additional tests. I think your approach of adding regressions as you find them is fine.

Reply:
Sorry, I didn't mean to imply that you had to write new test as part of this pull-request. I was merely thinking out loud about our lack of tests for scoping rules. I agree with you that we have it mostly right, and that's good enough for now. I also agree that adding regressions as you find them is a good way forward.